src/api/schema/gen.rs

```diff
@@ -0,0 +1,111 @@
+use std::fs;
+use vector::api::schema::build_schema;
+
+static INTROSPECTION_QUERY: &str = r#"
```

Comment:
So is this something we need to keep up to date when the graphql changes? Was this generated or written by hand?

Reply:
The query itself was hand-written, but it'll never change -- it always returns the latest schema at the time of running the query. We shouldn't ever have to touch this file, except to run it via:

```bash
cargo run --bin graphql-schema
```

The [graphql/schema.json](https://github.com/timberio/vector/blob/leebenson/graphql-client/graphql/schema.json) it generates _will_ change, and should be checked in. This is the same approach we're using in the [GraphQL branch in the UI](https://github.com/timberio/vector-ui/blob/lee/graphql/packages/vortex/src/graphql/index.tsx).

In Alloy, we had a GH Action to assert the checked-in version matches the current API. I'd anticipate doing the same here, so the two should never be out of sync.